### ISSUE TITLE
Wrong intellisense for SQL

### ISSUE THREAD

**Issue by EntwicklerMQ** [microsoft/vscode-mssql#17873]
Type: <b>Bug</b>

if you write "select * fro" intellisense doesn't suggest "from" as is it should do.  Sometimes its even worse while typing "fr" it suggests "for"

Extension version: 0.4.0
VS Code version: Code 1.85.0 (af28b32d7e553898b2a91af498b1fb666fdebe0c, 2023-12-06T20:48:09.019Z)
OS version: Windows_NT x64 10.0.22631
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i5-1235U (12 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.73GB (4.50GB free)|
|Process Argv|--crash-reporter-id f0bc4c54-a8c1-4950-bdb5-107e26225614|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263:30899288
vsclangdf:30486550
c4g48928:30535728
dsvsc012:30540252
azure-dev_surveyone:30548225
3biah626:30602489
89544117:30613380
showlangstatbar:30737416
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30902429
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:30885898
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
aa_t_chat:30882232
dsvsc019:30917259

```

</details>

**Comment by kisantia** [comment]
@EntwicklerMQ transferred the issue to the vscode-mssql repo

**Comment by EdCallahan** [comment]
I get FROM as a second option. Same version, v1.22.1. Strange.

![image](https://github.com/microsoft/vscode-mssql/assets/42499434/7f08dfb6-3c28-40c1-94e7-9741077cf5d4)

**Comment by kburtram** [comment]
I'm also not seeing this behavior when typing the repro script provided in the initial description.  The reported extension version is `0.4.0`, which I suspect is a typo since that would be a very old release.  

Could you confirm this behavior with the 1.23.0 release?  Any other details that may help to reproduce this behavior?

![image](https://github.com/user-attachments/assets/92571580-8392-4cce-9f20-0e5063e496e8)

**Comment by kburtram** [comment]
I've been able to repro this behavior in the scenario where the term "fro" is edited with backspacing & retyping with various mixes of 'f', 'r', and 'o' characters, though I'm not sure if that is what's being reported here.

The VS Code intellisense completion list works basically as described, at some point the keyword list is requested and all SQL keywords are returned from the extension to vscode.  This completion list is maintained in VS Code (outside the extension) and is filtered based on what the user has typed.  The filtering matches any of the characters that display in the same order in both works, e.g., 'fr' matches 'for' since the two character are in both terms in the same order.  The filtering done incrementally, so it's possible to get the filtered completion list into a state where further edits are only matching against the already filtered set of terms, thus leading to behavior similar to what is shown in the initial screenshoot.  To get back to the full keyword list in that case you'd need to delete the characters back to the blank space to reset the completion list set.

The above behavior is controlled by VS Code and can't really be controlled by the extension.  If you're not seeing 'FROM' when typing 'FRO' without backspacing and editing, that would be unexpected and I'm not able to reproduce that without further details.

**Comment by EntwicklerMQ** [comment]
> I'm also not seeing this behavior when typing the repro script provided in the initial description. ~The reported extension version is `0.4.0`, which I suspect is a typo since that would be a very old release.~ (nevermind, I see the second comment corrects the extension version details).
> 
> Could you confirm this behavior with the 1.23.0 release? Any other details that may help to reproduce this behavior?
> 
> ![image](https://private-user-images.githubusercontent.com/599935/358705431-92571580-8392-4cce-9f20-0e5063e496e8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMjk5ODAsIm5iZiI6MTcyNDIyOTY4MCwicGF0aCI6Ii81OTk5MzUvMzU4NzA1NDMxLTkyNTcxNTgwLTgzOTItNGNjZS05ZjIwLTBlNTA2M2U0OTZlOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMVQwODQxMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMzJjNmI3ZDRkZGE1NDUzMGU4ZTBlZjQyZjFlNWZjNzJiODJiODJkMjgwN2M5NmYzNzZkOWU5ZTU4MWUxZWU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.bduCCrGyR64d9tH5WbWoMCghmqBrYCZOORAra3FuoYk)

Yes, I can confirm that this behaviour is still comming up with the version 1.23 - no backspacing is involved

**Comment by kburtram** [comment]
@EntwicklerMQ thanks for the update.  Unfortunately, I'm unable to reproduce this issue.  Please let me know if you have any additional suggestions on how to reproduce.  Do you possibly have any other SQL related extensions installed that could be conflicting with MSSQL suggestions?

**Comment by aasimkhan30** [comment]
Closing as ordering of individual completion items is decided by vscode.
